data-products/src/main/scala/org/sunbird/analytics/audit/ActivityAggMigrationJob.scala

```diff
@@ -0,0 +1,85 @@
+
+import com.datastax.spark.connector.cql.CassandraConnectorConf
+import org.apache.spark.sql.cassandra.CassandraSparkSessionFunctions
+import org.apache.spark.sql.expressions.Window
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.{SparkSession, _}
+import org.apache.spark.{SparkContext, sql}
+import org.ekstep.analytics.framework.conf.AppConf
+import org.ekstep.analytics.framework.util.{CommonUtil, JSONUtils, JobLogger}
+import org.ekstep.analytics.framework.{FrameworkContext, IJob, JobConfig}
+import org.sunbird.analytics.job.report.BaseReportsJob
+
+object ActivityAggMigrationJob extends optional.Application with IJob with BaseReportsJob {
+  val userActivityAggDBSettings = Map("table" -> "user_activity_agg", "keyspace" -> AppConf.getConfig("sunbird.user.report.keyspace"), "cluster" -> "LMSCluster")
+  val assessmentAggregatorDBSettings = Map("table" -> "assessment_aggregator", "keyspace" -> AppConf.getConfig("sunbird.user.report.keyspace"), "cluster" -> "LMSCluster")
+  val cassandraUrl = "org.apache.spark.sql.cassandra"
+  implicit val className: String = "org.sunbird.analytics.audit.ActivityAggMigrationJob"
+  val jobName = "ActivityAggMigrationJob"
+  // $COVERAGE-OFF$ Disabling scoverage for main and execute method
+  override def main(config: String)(implicit sc: Option[SparkContext], fc: Option[FrameworkContext]): Unit = {
+    JobLogger.init(jobName)
+    JobLogger.start(s"$jobName started executing", Option(Map("config" -> config, "model" -> jobName)))
+
+    implicit val jobConfig: JobConfig = JSONUtils.deserialize[JobConfig](config)
+    implicit val spark: SparkSession = openSparkSession(jobConfig)
+    implicit val frameworkContext: FrameworkContext = getReportingFrameworkContext()
+    spark.setCassandraConf("LMSCluster", CassandraConnectorConf.ConnectionHostParam.option(AppConf.getConfig("sunbird.courses.cluster.host")))
+    try {
+      val res = CommonUtil.time(migrateData(spark))
+      updatedTable(res._2, userActivityAggDBSettings)
+      JobLogger.end(s"$jobName completed execution", "SUCCESS", Option(Map("timeTaken" -> res._1, "totalRecordsUpdated" -> res._2)))
+    } finally {
+      frameworkContext.closeContext()
+      spark.close()
+    }
+  }
+  // $COVERAGE-ON$
+  def migrateData(session: SparkSession): DataFrame = {
+    val mergeAggCol = session.udf.register("agg_latest", updateAggCol())
+    val mergeAggLastUpdatedCol = session.udf.register("agg_last_updated_latest", updateAggLastUpdatedCol())
+    val activityAggDF = fetchActivityData(session, fetchData)
+    val assessmentAggDF = getBestScoreRecordsDF(fetchAssessmentData(session, fetchData))
+
+    val filterDF = activityAggDF.join(assessmentAggDF, activityAggDF.col("activity_id") === assessmentAggDF.col("course_id") &&
+      activityAggDF.col("userid") === assessmentAggDF.col("user_id") &&
+      activityAggDF.col("context_id") === assessmentAggDF.col("batchid"), joinType = "inner")
```

Comment:
context_id value is "cb:<batchId>"

Reply:
I have handled this "cb:" string concat logic at the `getBestScoreRecordsDF` method already.